### ISSUE TITLE
PDF Highlights Doubling Up

### ISSUE THREAD

**Issue by rmuser2** [koreader/koreader#8084]
* KOReader version: Latest Stable
* Device: Remarkable 2

#### Issue
A PDF is synced to the RM, I read and highlight it using KOReader. I exit the PDF, KOReader asks me to save the changes. I have highlights set to be written to the PDF.

This syncs back to my drive and Acrobat can delete the KOReader highlights, which is great.

When I open the PDF on KOReader again, it seems that the highlights are reapplied to each section, making the highlight darker. On one paragraph, it got to such a point that it seemed like it was redacted. 

It looks like the highlight which is written to a paragraph is saved in the metadata and then reapplied to the same paragraph. I can delete the metadata, leaving the "hard written" original highlight underneath it all.

#### Steps to reproduce
Setup KOReader to write highlights to the PDF and do some back and forth syncing, highlighting a different passage on KOReader each time.

**Comment by rmuser2** [comment]
I think the paragraph which was completely blacked out may be a one off. I have created a test document and sent it across to the RM2 and back, highlighting as I go. At most, RM is showing a second highlight layer after a highlight has been written to the PDF. On successive saves, Acrobat still sees this as one highlight and it does not cause the test to be buried underneath the yellow highlighting, so the issue is just on the KOReader side which is not as bad as I thought.

**Comment by poire-z** [comment]
Yes, that's unfortunately how it works (we write it into the pdf but also keep it on our side - then, we get the native pdf ones rendered by MuPDF + the one we draw over... hard to know if they are the same and avoid drawing ours (dunno, may be not, but you'd have yellow natives ones instead of grey ones - anyway, I'm done with PDF highlights hell, tried to be nice, but that's really not my war :).
See https://github.com/koreader/koreader/issues/7928#issuecomment-879859250 for details.

Also, some recent fixes related to PDF highlights in https://github.com/koreader/koreader/pull/7988 may solved your original issue - will be in next stable.
